Components: Links in notices are not underlined

psealock opened the pull request:
**Issue**
When the user is presented with a notice, links are missing an underline. The problem is addressed in issue https://github.com/Automattic/wp-calypso/issues/4321.

**Steps to reproduce**
1. Try to add a new domain to your site.
2. Enter cnn.com in the search domain field. 
3. A notice should show up saying:
>cnn.com is taken.
If you purchased cnn.com elsewhere, you can map it for €13.00.

The ```map it``` part is the link that is missing the underline.

**What should happen**

![screen shot 2016-04-01 at 11 06 42 am](https://cloud.githubusercontent.com/assets/1922453/14192415/04b3eb0a-f7fa-11e5-8e83-7422611220fc.png)

**What does happen** - Chrome 49 (retina and normal), Safari 9.1 (retina and normal)

![screen shot 2016-04-01 at 11 06 59 am](https://cloud.githubusercontent.com/assets/1922453/14192424/12c09a72-f7fa-11e5-9003-67d64b0d7e1e.png)

**Notes**
Why not use ```text-decoration: underline```? Browser support is [really bad!](http://caniuse.com/#search=text-decoration). There are several approaches to solve the problem, outlined [in a nice blog post](http://benjam.info/blog/posts/2015-01-31-css-underline) with ```linear-gradient``` as a background image being the most suited to this situation. 

**Fix**
* The underline was placed too far down, so it wasn't showing up. I made it exactly the ```line-height```
* The height of the background needs to take into account retina display.

**Tested**
* Chrome 49 (retina, normal)
* Safari 9.1 (retina, normal)
* Firefox 44 (retina, normal)
* IE 11 using a this [code pen](http://codepen.io/jxnblk/pen/jCmwK/) (retina, normal)

ping @mtias @klimeryk

klimeryk commented:
Awesome contribution (and a very well written description)! :+1: 
I can confirm this has indeed fixed the issue I've reported. I'll let @mtias give the final green light though, as he is more suited to verify the fix from technical point of view.

psealock commented:
**Update**
The linear gradient used different percentages for start and stop, creating the need for a media query for retina displays to change the ```background-size```. After some playing around, using the same start/stop of 50% eliminates the need for the media query altogether.

**Changes**
* Change the linea-gradient's start/stop to 50%.
* Remove media query for retina displays.
* Change width to ```2px```.
* Rebased and squashed commits.

**Before** (non-retina)

![screen shot 2016-04-01 at 12 33 16 pm](https://cloud.githubusercontent.com/assets/1922453/14194110/3d15a54e-f807-11e5-900a-99a891d5e783.png)

**After** (non-retina)

![screen shot 2016-04-01 at 12 32 28 pm](https://cloud.githubusercontent.com/assets/1922453/14194118/4c8533c8-f807-11e5-9252-ca5351906eea.png)


**Tested**
* Chrome 49 (retina, normal)
* Safari 9.1 (retina, normal)
* Firefox 44 (retina, normal)
* IE 11 using a this [code pen](http://codepen.io/jxnblk/pen/jCmwK/) (retina, normal)

@klimeryk , sorry to make you take another look, but its much cleaner this way.

klimeryk commented:
Sweet! Less code, same effect - me likey :)
Tested on Linux (normal, Chrome and Firefox) and OSX (retina, Safari and Chrome), working nicely :+1:

klimeryk commented:
Pinging @mtias again - this seems like a nice and clean solution to me, but I'd prefer if someone more knowledgeable in CSS would chip in here.

mtias commented:
Ugh, sorry I missed this. @rickybanister and @shaunandrews recently worked on the underlining of notices links as well, so looping them in.

shaunandrews commented:
I think this will be fixed in #4822

MichaelArestad commented on client/components/notice/style.scss:
```diff
@@ -137,15 +137,10 @@
 
 		// Medium text underline hack
-		background-image: linear-gradient(to bottom, rgba(0,0,0,0) 75%, #fff 50%);
+		background-image: linear-gradient(to bottom, rgba(0,0,0,0) 50%, #fff 50%);
```
This won't produce a single pixel line on hidpi screens. If we can't do that, we might as well go with `text-decoration: underline`

mtias commented:
This should be fixed now.
